test/rules/strict-boolean-expressions/allow-string/test.ts.lint

```diff
@@ -0,0 +1,14 @@
+declare function get<T>(): T;
+
+if (get<string>()) {}
+
+if (get<boolean | string>()) {}
+    ~~~~~~~~~~~~~~~~~~~~~~~ [If statement condition mixes multiple kinds of falsy values. Expected a boolean or string.]
```

Comment:
`This type is not allowed in the 'if' condition because it unions one or more disallowed truthy/falsy types. Allowed types are boolean or string